token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -586,70 +554,123 @@ fn process_transfer(
-        ));
-        let source_transfer_amount_hi = EncryptedBalance::from((
-            proof_data.ciphertext_hi.commitment,
-            proof_data.ciphertext_hi.source_handle,
-        ));
+        let source_transfer_amount_lo =
+            transfer_amount_source_ciphertext(&proof_context.ciphertext_lo);
+        let source_transfer_amount_hi =
+            transfer_amount_source_ciphertext(&proof_context.ciphertext_hi);
 
         process_source_for_transfer(
             program_id,
             source_account_info,
             mint_info,
             authority_info,
             account_info_iter.as_slice(),
-            &proof_data.transfer_with_fee_pubkeys.source_pubkey,
+            &proof_context.transfer_with_fee_pubkeys.source,
             &source_transfer_amount_lo,
             &source_transfer_amount_hi,
-            &proof_data.new_source_ciphertext,
+            &proof_context.new_source_ciphertext,
             new_source_decryptable_available_balance,
         )?;
 
         // From the proof data decode lo and hi transfer amounts encrypted under the destination
         // encryption public key
-        let destination_transfer_amount_lo = EncryptedBalance::from((
-            proof_data.ciphertext_lo.commitment,
-            proof_data.ciphertext_lo.destination_handle,
-        ));
-        let destination_transfer_amount_hi = EncryptedBalance::from((
-            proof_data.ciphertext_hi.commitment,
-            proof_data.ciphertext_hi.destination_handle,
-        ));
+        let destination_transfer_amount_lo =
+            transfer_amount_destination_ciphertext(&proof_context.ciphertext_lo);
+        let destination_transfer_amount_hi =
+            transfer_amount_destination_ciphertext(&proof_context.ciphertext_hi);
 
         process_destination_for_transfer(
             destination_token_account_info,
             mint_info,
-            &proof_data.transfer_with_fee_pubkeys.destination_pubkey,
+            &proof_context.transfer_with_fee_pubkeys.destination,
             &destination_transfer_amount_lo,
             &destination_transfer_amount_hi,
-            Some((&proof_data.fee_ciphertext_lo, &proof_data.fee_ciphertext_hi)),
+            Some((
+                &proof_context.fee_ciphertext_lo,
+                &proof_context.fee_ciphertext_hi,
+            )),
         )?;
     }
 
     Ok(())
 }
 
+/// Extract the transfer amount ciphertext encrypted under the source ElGamal public key.
+#[cfg(feature = "zk-ops")]
+fn transfer_amount_source_ciphertext(
+    transfer_amount_ciphertext: &TransferAmountCiphertext,
+) -> ElGamalCiphertext {
+    let transfer_amount_ciphertext_bytes = bytemuck::bytes_of(transfer_amount_ciphertext);
+
+    let mut source_ciphertext_bytes = [0u8; 64];
+    source_ciphertext_bytes[..32].copy_from_slice(&transfer_amount_ciphertext_bytes[..32]);
+    source_ciphertext_bytes[32..].copy_from_slice(&transfer_amount_ciphertext_bytes[32..64]);
```

Comment:
nit: This is definitely done purposefully  -- can you add comments about specifically which bits you're extracting for these lines?

Reply:
Done!